### PR TITLE
[impala] improved query timeout error logging

### PR DESCRIPTION
**Pull request by amitsrivastava**
we were not clearly catching the timeout error from impala query submission. the error response was getting passed to the upper layers that errored out when trying to get json out of it. It lead to a cryptic error on the frontend as well. Now we are generating a clearer log message with the timeout values and config name. The frontend also shows a timeout error message.

### CODE REVIEW

File: desktop/core/src/desktop/lib/thrift_util.py [hunk 1]
```diff
@@ -557,6 +557,14 @@ def wrapper(*args, **kwargs):
...
         except Exception as e:
           logging.exception("Thrift saw exception (this may be expected).")
+          if "Read timed out." in str(e):
+            raise StructuredException(
+              'TIMEOUT',
+              'Current timeout of %s was hit. Please consider increasing the `server_conn_timeout` config. Request failed with "%s"' %
+              (self.timeout_seconds, str(e)),
+              data=None,
+              error_code=502
+            )
```
Comment (Harshg999) [inline desktop/core/src/desktop/lib/thrift_util.py:567]: Nice!

**Review by Harshg999 — APPROVED** [review]
LGTM!

File: desktop/core/src/desktop/lib/thrift_util.py [hunk 2]
```diff
@@ -557,6 +557,14 @@ def wrapper(*args, **kwargs):
               raise
         except Exception as e:
           logging.exception("Thrift saw exception (this may be expected).")
+          if "Read timed out." in str(e):
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/lib/thrift_util.py:560]: Is this string specific to impala? If so, perhaps see if we can include other engines as well?

**Review by JohanAhlen — APPROVED** [review]
Nice!

File: desktop/core/src/desktop/lib/thrift_util.py (hunk 2 above)
Comment (amitsrivastava) [reply desktop/core/src/desktop/lib/thrift_util.py:560]: We are getting a read timeout from Impala. This string is actually coming from our underlying REST client library, so it should be engine agnostic.

Comment (JohanAhlen) [reply desktop/core/src/desktop/lib/thrift_util.py:560]: Excellent.

**Pull request merged**